### DISCUSSION TITLE
How to prevent creating variable with name `$` ?

### DISCUSSION THREAD

**Discussion by CursedMun** [swc-project/swc#9648] [Q&A]
How can I prevent the creation of variables with the name $ when using mangle and compress?

Here's an example that illustrates the issue where a variable named $ is created.

[swc playground](https://play.swc.rs/?version=1.7.35&code=H4sIAAAAAAAAA2XSzWrCUBDF8X2e4iwVSsk58Sv0PbpPxRYhJKBtN6XvXp0zE5C6mLkI98eVv%2B9f0%2FHzPE8Y54%2FX4XIe3sbTddV8Dxc%2B3adidjE3MbcxdzH3MQ8x%2B5hsvXybvk7fpwFaoAnaoBFakRXlG6zIiqzIiqzIiqzISpdvaZPJ3eXe5N7m3uXe5z7k7nOzrUOJLJJlslCWymJZLgtmv7ysDstbS1bJKlklq2SVrJLVL7%2F1fljjpwGO83Sdx9PzrfQq%2BiL6Ivoi%2BiL6Ivoi%2BiL6Ivoi%2B%2BLh49pwbbg2XBuuDdeGa8O14drI2v9M5ftsyqZsyqZsyqZsymbXrl%2Ba36Z5%2BFvfvvoDUn78RPACAAA%3D&config=H4sIAAAAAAAAA41VS47bMAzdzykCocsuZlq0KHqA7nqCQSEoFuVoKkuGSHsSBLl7adnOZ0wb3QQxnx4pko%2FU%2BWm3U29YqZ%2B7M%2F%2Flj9ZkhHz9ZgueIpkjWxSdWsAq%2B5bU5xklHCBnAkIxXUZEkck10MAC%2FPL88m1iqJASwsyYbI2P3p3uY1apaTMg3tnYyi67BiLhI3%2FCcnofAMrdvX2fUgATNxBtUPtIUEOWHFcpBNMi6N5kwctwU5M9JinEAHYEVrc5tSIerSefIsdcohaM1VWyIEA%2BQ0W%2BB4nGsZgWkdMT8imwhX1X16XPH9jQm9AZEmLCsbSEbyt4PSSPpF0XpRKO4EoNRnAq7kemdzoDdTkueW%2FJx5We%2FAXgCgSDGE0Dkt9ywrGe1thuk%2BmjY8nSScBZ31KWEWouqvbeCZUdKgOZvNTNDLarYKhsJV1nglfKh96CBudYK4JrfPdUHaSgw6AnJwDcX%2BMkVY2Avk7hCj4MxAb8i7MkWWDTicbQYR3FU7NPYSNAA3RIduMAt4LSOpx5SxzbdbyLFlgaYMUjHRZguQR4ACjpUPblQhs8HuxR1yHtxTXRss5LE79Otst1OTcm1mUJnGWx3cxFSrz4%2B3LDV%2FVJ%2FbmFuDwIbu2UotQG6CGsTc1%2FTOQmyqXohyGaF8liWcAPaQYMz7l%2FeX54pTinp%2Fm3ZKeaZLu7UpURGF%2Bv7%2Bp2aH6o5hYoj79n4uj98g%2F8o0LlUwcAAA%3D%3D)

Why is this a problem? When minifying for the web, pages that use jQuery depend on the $ symbol. If a variable named $ is created, it can conflict with jQuery's functionality.

How can I prevent this from happening when using mangle and compress?

**Comment by kdy1** [comment]
`reserved: ["$"]` should do the trick, but IIRC the implementation does not check `reserved` if there's no var named it.
It's a bug. Can you file an issue?

A quick workaround is adding `var $ = '';` to the end of the file and removing it from the output using string replacement.



https://play.swc.rs/?version=1.7.35&code=H4sIAAAAAAAAA2XSzWrCUBDF8f19irMQVJCSOfErlL5G91FsEUIC2nZT%2Bu5N5swERBczVyE%2Frv79%2BO7PX9ehRzd8vre3a3vqLvdV%2BWlvtpkmfdY%2Btz53Pvc%2BDz6PPhufVmnpadPjpudNgEkwESbDhJgUSmHcQQqlUAqlUAqlUAql1HGXKpjYdext7F3sfexD7GPsJrZVeUjRkrQ0LVFL1ZK1dC1ha%2Bab5WG%2Ba8pMmSkzZabMlJkym%2Fm7Toc1fgtwHvr70F1extIr7wvvC%2B8L7wvvC%2B8L7wvvC%2B%2BL6IuHl2pDtaHaUG2oNlQbqg3Vhmojaj%2BZjPvJpEzKpEzKpEzKpMy6Wr%2BWv1Ie%2FtbjR9OPggXesFyOb%2F4BV2Frdv4CAAA%3D&config=H4sIAAAAAAAAA41VS47bMAzdzykCocsuZlq0KHqA7nqCQSEoFuVoKkuGSHsSBLl7adnOZ0wb3QQxnx4pko%2FU%2BWm3U29YqZ%2B7M%2F%2Flj9ZkhHz9ZgueIpkjWxSdWsAq%2B5bU5xklHCBnAkIxXUZEkck10MAC%2FPL88m1iqJASwsyYbI2P3p3uY1apaTMg3tnYyi67BiLhI3%2FCcnofAMrdvX2fUgATNxBtUPtIUEOWHFcpBNMi6N5kwctwU5M9JinEAHYEVrc5tSIerSefIsdcohaM1VWyIEA%2BQ0W%2BB4nGsZgWkdMT8imwhX1X16XPH9jQm9AZEmLCsbSEbyt4PSSPpF0XpRKO4EoNRnAq7kemdzoDdTkueW%2FJx5We%2FAXgCgSDGE0Dkt9ywrGe1thuk%2BmjY8nSScBZ31KWEWouqvbeCZUdKgOZvNTNDLarYKhsJV1nglfKh96CBudYK4JrfPdUHaSgw6AnJwDcX%2BMkVY2Avk7hCj4MxAb8i7MkWWDTicbQYR3FU7NPYSNAA3RIduMAt4LSOpx5SxzbdbyLFlgaYMUjHRZguQR4ACjpUPblQhs8HuxR1yHtxTXRss5LE79Otst1OTcm1mUJnGWx3cxFSrz4%2B3LDV%2FVJ%2FbmFuDwIbu2UotQG6CGsTc1%2FTOQmyqXohyGaF8liWcAPaQYMz7l%2FeX54pTinp%2Fm3ZKeaZLu7UpURGF%2Bv7%2Bp2aH6o5hYoj79n4uj98g%2F8o0LlUwcAAA%3D%3D

**Reply by CursedMun** [reply]
Hello, thanks for the quick response. I'll use a temporary workaround for now, and I have created an [issue](https://github.com/swc-project/swc/issues/9650).
